core/frontend/src/main/scala/io/udash/properties/Property.scala

```diff
@@ -117,6 +140,74 @@ trait ReadableProperty[A] {
   }
 }
 
+abstract class BaseReadableSeqPropertyFromSingleValue[A, B : ModelValue](origin: ReadableProperty[A], transformer: A => Seq[B],
+                                                                         override val executionContext: ExecutionContext)
+  extends ReadableSeqProperty[B, ReadableProperty[B]] {
+
+  override val id: UUID = UUID.randomUUID()
+  override protected[properties] val parent: Property[_] = null
+  protected val structureListeners: mutable.Set[Patch[Property[B]] => Any] = mutable.Set()
+
+  val pc = implicitly[PropertyCreator[B]]
+  protected val children = mutable.ListBuffer.empty[Property[B]]
+
+  update(origin.get)
+  origin.listen(update)
+
+  private def structureChanged(patch: Patch[Property[B]]): Unit =
+    structureListeners.foreach(_.apply(patch))
+
+  private def update(v: A): Unit = {
+    val transformed = transformer(v)
+    val current = get
+    val commonBegin = {
+      var tmp = 0
+      while (tmp < current.size && tmp < transformed.size && current(tmp) == transformed(tmp)) tmp += 1
+      tmp
+    }
+    val commonEnd = {
+      var tmp = 0
+      while (0 < current.size - tmp && 0 < transformed.size - tmp
+        && current(current.size - tmp - 1) == transformed(transformed.size - tmp - 1)) tmp += 1
+      tmp
+    }
+
+    val patch = if (transformed.size > current.size) {
+      val added: Seq[CastableProperty[B]] = Seq.fill(transformed.size - current.size)(pc.newProperty(this)(executionContext))
+      children.insertAll(commonBegin, added)
+      Patch(commonBegin, Seq(), added, false)
+    } else if (transformed.size < current.size) {
+      val removed = children.slice(commonBegin, commonBegin + current.size - transformed.size)
+      children.remove(commonBegin, current.size - transformed.size)
+      Patch(commonBegin, removed, Seq(), transformed.isEmpty)
+    } else null
+
+    CallbackSequencer.sequence {
+      transformed.zip(children)
+        .slice(commonBegin, math.max(commonBegin + transformed.size - current.size, transformed.size - commonEnd))
+        .foreach { case (pv, p) => p.set(pv) }
+      if (patch != null) structureChanged(patch)
+      valueChanged()
+    }
+  }
+
+  override def get: Seq[B] =
+    children.map(_.get)
+
+  override def elemProperties: Seq[ReadableProperty[B]] =
```

Comment:
This implements a read-only view. I believe with this implementation it is possible to pattern match on the elements of returned sequence and modify them as regular Properties by mistake.